apollo-router/src/services/subgraph_service.rs

```diff
@@ -73,13 +75,23 @@ pub(crate) struct SubgraphService {
 }
 
 impl SubgraphService {
-    pub(crate) fn new(service: impl Into<String>) -> Self {
+    pub(crate) fn new(service: impl Into<String>, tls_cert_store: Option<RootCertStore>) -> Self {
         let mut http_connector = HttpConnector::new();
         http_connector.set_nodelay(true);
         http_connector.set_keepalive(Some(std::time::Duration::from_secs(60)));
         http_connector.enforce_http(false);
+        let tls_config = match tls_cert_store {
+            None => rustls::ClientConfig::builder()
+                .with_safe_defaults()
+                .with_native_roots()
+                .with_no_client_auth(),
+            Some(store) => rustls::ClientConfig::builder()
+                .with_safe_defaults()
+                .with_root_certificates(store)
```

Comment:
Is it intentional that the specified roots are used instead of the native roots, not in addition to? This seems inconvenient if only some of the subgraphs use self-signed certs.

Reply:
yes, it is intentional, because when overriding the root certificate list, you don't want other certificate authorities to be able to generate valid certificates